### PR TITLE
validate api version if available

### PR DESCRIPTION
**Pull request by LiNk-NY**
Here we use an `api_reference_version` to check against when running the `Service` class constructor.
It extracts the version from `info: version:` using the `yaml` package.

### CODE REVIEW

File: R/Service.R [hunk 1]
```diff
@@ -20,13 +20,24 @@ setOldClass("request")
...
 .config <- function(x) x@config
 
+.service_get_api_file <- function(reference_url, reference_headers) {
+    fl <- tempfile()
+    response <- GET(
+        reference_url,
+        add_headers(.headers = reference_headers),
+        write_disk(fl)
+    )
+    .avstop_for_status(response, ".service_validate_md5sum")
```
Comment (mtmorgan) [inline R/Service.R]: does the 2nd argument accurately reflect where the error is occurring?

**Review by mtmorgan — COMMENTED** [review]
I know they're difficult but any chance of a unit test, e.g., that the hard-coded `api_reference_version` on R/Dockstore:42 is still current? The unit test would fail noisily for the developer, rather than the user receiving a warning

File: R/Service.R (hunk 1 above)
Comment (LiNk-NY) [reply R/Service.R]: Thanks! I've resolved this in the latest change

**Comment by LiNk-NY** [comment]
> rather than the user receiving a warning

@mtmorgan 
Do you suggest to remove the `warning` within `.service_validate_version` / `.service_validate_md5sum` and remove from the `Service` constructor? and I only run those in the unit tests?

**Comment by mtmorgan** [comment]
I think the user should still get the warning, so that they know that they need to update something (or tell the maintainer that they should!); the unit test just makes it less likely that the maintainer hears about problems from the user...

**Pull request merged**